src/meta/server_state.h

```diff
@@ -291,6 +291,12 @@ class server_state
     void process_one_partition(std::shared_ptr<app_state> &app);
     void transition_staging_state(std::shared_ptr<app_state> &app);
 
+    void register_rpc_handlers();
+    void unregister_rpc_handlers();
+
+    // client => meta server
+    void on_query_configuration_by_index(configuration_query_by_index_rpc rpc);
```

Comment:
> all rpc_handlers in meta server is registered in meta_service. And then the rpc request is forwarded to server_state、bulk_load_service and so on

I think the previous structure is clear, only move `on_query_configuration_by_index` into server_state breaking the design, and I didn't see some merit in it

Reply:
Break what desgin? I think the main logic is implemented by `server_state` but not `meta_service`. The work of `meta_service` is forwarding. And there are a lot of other rpc handlers about `backup_service`、 `split_service`, and so on.

Reply:
> ...meta_service. The work of meta_service is forwarding

So my means is: on_xxx_() in meta_service forward to xxx() in server_state should be an appropriate design